lib/cretonne/src/isa/mod.rs

```diff
@@ -148,6 +148,14 @@ pub trait TargetIsa {
     /// This is just used for printing and parsing encodings in the textual IL format.
     fn recipe_names(&self) -> &'static [&'static str];
 
+    /// Get the index of a recipe name if it exists.
+    fn find_recipe_index(&self, recipe_name: &str) -> Option<u16> {
```

Comment:
I don't think this needs to go in `TargetIsa`. ISAs are not going to have very long lists of recipes, and the parser is not performance critical, so a linear search should be good enough for everybody. (We can always do this later, if it turns out I'm wrong.)

I would prefer this as a function in `parser.rs` instead.